core/src/main/java/org/mule/processor/AsyncDelegateMessageProcessor.java

```diff
@@ -119,18 +122,28 @@ else if (!(flowConstruct instanceof StageNameSourceProvider))
         }
     }
 
+    @Override
     public MuleEvent process(MuleEvent event) throws MuleException
     {
         if (event.isTransacted())
         {
             throw new MessagingException(CoreMessages.asyncDoesNotSupportTransactions(), event, this);
         }
 
+        final MuleMessage message = event.getMessage();
+        if (!consumablePayloadWarned && isConsumable(message.getPayload().getClass()))
+        {
+            consumablePayloadWarned = true;
```

Comment:
I don't agree with this behaviour. I think we should either consume the payload or fail (like scatter-gather does). Allowing this can open the door for race conditions without much knowledge about concurrency will have a hard time understanding.

I'm also pretty sure that somewhere down the line, the `<async>` scope is either validating or consuming the payload in this case. @pablolagreca  do you remember where that is? @vromero didn't you spend a lot of time with a race condition problem on the `<async>` as well? any feedback on this?